app/uk/gov/hmrc/gform/controllers/AuthenticatedRequestActions.scala

```diff
@@ -195,9 +197,17 @@ class AuthenticatedRequestActions(
     maybeAccessCode: Option[AccessCode],
     formTemplate: FormTemplate)(retrievals: MaterialisedRetrievals)(implicit hc: HeaderCarrier): Future[Result] =
     for {
-      form        <- gformConnector.getForm(FormId(retrievals, formTemplate._id, maybeAccessCode))
-      obligations <- obligationService.lookupObligationsMultiple(formTemplate)
-      result      <- f(AuthCacheWithForm(retrievals, form, formTemplate, obligations))
+      form    <- gformConnector.getForm(FormId(retrievals, formTemplate._id, maybeAccessCode))
+      newForm <- obligationService.lookupIfPossible(form, formTemplate, authService, retrievals)
+      update <- obligationService.updateObligations(
+                 form._id,
+                 UserData(newForm.formData, newForm.status, newForm.visitsIndex, newForm.obligations),
+                 form,
+                 newForm)
+      result <- {
+        update
```

Comment:
`update` ?

Reply:
The reason update was included here was to ensure that updateObligations actually finishes before moving on

Reply:
That is guaranteed by monad laws ie. by flatMap method.